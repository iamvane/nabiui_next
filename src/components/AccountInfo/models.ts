export enum Gender {
  male = 'male',
  female = 'female',
  undislcosed = 'undisclosed'
}

export enum VerificationChannel {
  Text = 'sms',
  Call = 'call'
}

export interface AccountInfoType {
  firstName: string;
  lastName: string;
  middleName?: string;
  gender?: Gender | string;
  phoneNumber?: string;
  location?: string;
  lat?: string;
  lng?: string;
  isPhoneVerified?: boolean;
}
