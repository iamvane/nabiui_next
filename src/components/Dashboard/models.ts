import { Request } from '../../redux/models/RequestModel';

interface StudentType {
  name: string;
  age: number;
}

export interface NextLessonType {
  id: number;
  date: string;
  time: string;
  timezone: string;
  studentDetails: StudentType;
  instructor: string;
}

export interface LessonType {
  bookingId: number;
  instrument: string;
  skillLevel: string;
  lessonsRemaining: number;
  parent?: string;
  studentName?: string;
  students?: StudentType[];
  age?: number;
  lastLessonId: number;
}

export interface Booking {
  applicationId: number;
  instrument: string;
  skillLevel: string;
  instructor: string;
  lessonsRemaining: number;
  students: StudentType[];
}

export interface MyRequestType {
  id: number;
  instrument: string;
  placeForLessons: string;
  requestMessage: string;
  requestTitle: string;
  skillLevel: string;
  lessonDuration: string;
  applications: number;
  studentDetails: StudentType[];
  createdAt: string;
}

export interface InstructorDashboardType {
  id: number;
  complete?: boolean;
  missingFields?: string[]
  backgroundCheckStatus: string;
  lessons: LessonType[];
  requests: Request[];
  nextLesson: NextLessonType;
}

export interface StudentLessonType {
  id: number;
  date: string;
  timezone: string;
  instructor: string;
  instructorId: number;
  status: string;
  grade: number;
  gradeComment: string;
}

export interface StudentDetailsType {
  id: number;
  name: string;
  instrument: string;
  nextLesson: NextLessonType;
  lessons: StudentLessonType[];
}

export interface ParentStudentDashboardType {
  students: StudentDetailsType[];
}
