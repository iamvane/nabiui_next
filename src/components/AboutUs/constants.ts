export interface Teammate {
  name: string;
  role: string;
  image: string;
  bio: string;
  linkedIn?: string;
  email?: string;
}

export const team: Teammate[] = [
  {
    name: 'Vanessa Charles',
    role: 'Founder & CEO/CTO',
    image: 'https://nabimusic.s3.us-east-2.amazonaws.com/assets/images/vanessa-charles.JPG',
    bio: 'Vanessa was born in Caracas, Venezuela, and grew up in Carupano, a small city on the Venezuelan Caribbean coast. She moved to Boston to attend college, and got her Bachelor’s from UMass Boston. Vanessa started Nabi with the dream of bringing music education to the lives of many children. She is a former classical piano teacher and self-taught software engineer. She worked at Oracle as a Software Engineer for almost two years, and helped her brothers create Coco Mercado, Venezuela’s first and only online supermarket. Vanessa means butterfly in Greek and Nabi means butterfly in Korean.',
    linkedIn: 'https://www.linkedin.com/in/vanessacharlesnabi/',
    email: 'vanessa@nabimusic.com'
  },
  {
    name: 'Jennie Legary',
    role: 'Marketing Strategist',
    image: 'https://nabimusic.s3.us-east-2.amazonaws.com/assets/images/jennie-legary.jpeg',
    bio: 'Jennie is a mezzo-soprano opera singer with a Master in Performing Arts Management from Accademia Teatro alla Scala. She’s also a design, marketing, and business consultant, with experience in promoting classical music, real estate, and early-stage startups. With her two other singer colleagues, she has co-founded Coloratura: The Opera Dolls, a company that produces educational musical toys for children.'
  },
  {
    name: 'Luis Llave',
    role: 'Back-end Engineer',
    image: 'https://nabimusic.s3.us-east-2.amazonaws.com/assets/images/luis-llave.png',
    bio: 'Luis was born in Arequipa (Perú), but was living in Venezuela, where graduated as an Electrical Engineer at University of Carabobo (UC). At UC, he worked in development of a web system for student academic records and class scheduling. Also, Luis helped build English-Spanish translations in Mandriva and Mageia Linux distributions. He returned to his natal city and now can see a volcano (Misti) everyday. He\'s a fan of Morricone music too.'

  },
  {
    name: 'Marycarmen Mora',
    role: 'Graphic Designer',
    image: 'https://nabimusic.s3.us-east-2.amazonaws.com/assets/images/marycarmen-mora.jpg',
    bio: 'Marycarmen is a licensed Graphic Designer. She got her education from the University of Antonio José de Sucre and from the University of Los Andes in Venezuela. She currently lives in Mérida and works as a graphic designer and illustrator. During her free time, she likes to do aerial silk and crossfit. Marycarmen is passionate about music.'
  },
  {
    name: 'Mariana Del Rosal',
    role: 'Content Writer',
    image: 'https://nabimusic.s3.us-east-2.amazonaws.com/assets/images/mariana-del-rosal.jpeg',
    bio: 'Mariana studied Language and Literature at the University of Buenos Aires. She is a part-time middle school teacher and has a great time teaching teens. She lives in Buenos Aires, Argentina, with her husband, a music teacher, and their two young kids, who are growing up surrounded by all kinds of musical instruments. Mariana enjoys writing more than anything, and  in her free time, she\'s learning to play the Irish tin whistle.'
  }
];

export const titleAbout = 'About Nabi Music';
export const ourMission = 'Our mission is to make music education easily accessible and bring encouragement to every young musician.';
export const titleWhatWeDo = 'What We Do';
export const whatWeDo = 'Nabi Music is an innovative music learning experience for children. Nabi allows parents to connect with qualified instructors worldwide and provides unique features that motivate parents’ involvement in their children musical path.'
export const titleTeam = 'The Team';
