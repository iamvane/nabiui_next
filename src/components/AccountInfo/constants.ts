import { VerificationChannel } from './models';

export namespace AccountInfoComponent {
  export enum SectionTitles {
    Gender = 'Gender',
    Avatar = 'Profile Image',
    Location = 'Location'
  }

  export enum Labels {
    Gender = 'Gender',
    Female = 'Female',
    Male = 'Male',
    Birthday = 'Birthday'
  }

  export enum Placeholders {
    FirstName = 'First Name',
    LastName = 'Last Name',
    MiddleName = 'Middle Name',
  }

  export enum FieldKey {
    FirstName = 'firstName',
    LastName = 'lastName',
    MiddleName = 'middleName',
    Gender = 'gender',
    PhoneNumber = 'phoneNumber',
    Token = 'token',
    Location = 'location'
  }

  export const FieldNames = {
    [FieldKey.FirstName]: 'firstName',
    [FieldKey.LastName]: 'lastName',
    [FieldKey.MiddleName]: 'middleName',
    [FieldKey.Gender]: 'gender',
    [FieldKey.PhoneNumber]: 'phoneNumber',
    [FieldKey.Token]: 'token'
  };

  export interface Errors {
    [FieldKey.FirstName]?: string;
    [FieldKey.LastName]?: string;
    [FieldKey.MiddleName]?: string;
    [FieldKey.Gender]?: string;
    [FieldKey.PhoneNumber]?: string;
    [FieldKey.Token]?: string;
    [FieldKey.Location]?: string;

  }

  export const firstNameMinVal = 1;
  export const firstNameMaxVal = 50;

  export const middleNameMaxVal = 50;

  export const lastNameMinVal = 1;
  export const lastNameMaxVal = 50;

  export const errorMessages = {
    [FieldKey.FirstName]: {
      emptyFirstName: 'Enter your first name, up to {maxFirstNameValue} characters.'.replace(
        '{maxFirstNameValue}',
        firstNameMaxVal.toLocaleString()
      ),
      invalidFirstName: 'Enter a valid first name.'
    },
    [FieldKey.MiddleName]: {
      emptyMiddleName: 'Enter your middle name, up to {maxMiddleNameValue} characters.'.replace(
        '{maxMiddleNameValue}',
        middleNameMaxVal.toLocaleString()
      ),
      invalidMiddleName: 'Enter a valid middle name.'

    },
    [FieldKey.LastName]: {
      emptyLastName: 'Enter your last name, up to {maxLastNameValue} characters.'.replace(
        '{maxLastNameValue}',
        lastNameMaxVal.toLocaleString()
      ),
      invalidLastName: 'Enter a valid last name.'
    },
    [FieldKey.Token]: {
      invalidVerificationCode: 'Invalid verification code.',
      enterYourVerificationCode: 'Enter verification code.'
    },
    [FieldKey.Gender]: 'Select an option.',
    PhoneNumberEmpty: 'Enter a valid phone number.',
    PhoneNumberNotVerified: 'Verify phone number.',
    PhoneMissingCountryCode: 'Select country code.',
    [FieldKey.Location]: 'Enter a valid location.'
  };
}

export namespace PhoneValidationComponent {
  export const usPhoneNumberLenght = 10;
}

export namespace PhoneValidationFormComponent {
  export const sectionTitle = 'Verify Phone Number';

  export enum FieldNames {
    PhoneNumber = 'phoneNumber',
    PhoneToken = 'token',
    VefificationChannel = 'verificationChannel'
  }

  export enum Placeholders {
    PhoneNumber = 'Phone Number',
    PhoneToken = 'Verification Token'
  }

  export enum Buttons {
    Verify = 'Request Verification',
    Validate = 'Verify Phone',
    Cancel = 'Cancel'
  }

  export const resendCode = `Resend code or change phone number`;

  export const validationSucessful = 'Phone Verified';

  export const verificationMethod = {
    Text: {
      label: 'Text',
      value: VerificationChannel.Text
    },
    Call: {
      label: 'Call',
      value:  VerificationChannel.Call
    }
  };
}

export namespace LocationComponent {
  export const sectionTitle = 'Location';
  export const description = 'Please enter your home address.';
}

export namespace MiddleNameModalComponent {
  export const title = 'Missing Middle Name';
  export const message = 'Looks like you didnâ€™t provide a middle name. Please enter it below, if applicable.';

  export const noMiddleName = `I don't have one`;
  export const applyButton = 'Apply';
}
