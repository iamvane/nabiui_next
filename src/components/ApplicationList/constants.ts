
export namespace ApplicationListComponent {
  export const breadcrumbLabels = {
    home: 'Home',
    applications: 'Applications'
  };

  export const pageTitle = 'Applications';
}


export namespace ApplicationCardComponent {
  export const availabilitySection = 'Availability';
  export const messageSection = 'Message';
  export const rate = 'Rate:'
  export const bookLessons = 'Book Lessons'
}
