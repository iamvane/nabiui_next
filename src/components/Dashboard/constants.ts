export namespace DashboardComponent {
  export const pageTitle = 'Dashboard';
}

export namespace InstructorDashboardComponent {
  export const myStudents = 'My Students';
  export const noStundets = 'You currently have no students';
  export const findJobsButton = 'Find Jobs';
  export const applyToJobs = 'Apply to Jobs Near You';
  export const viewAll = 'View all';
}

export namespace StudentCardComponent {
  export const studentLabel = 'Student';
  export const ageLabel = 'Age';
  export const instrumentLabel = 'Instrument';
  export const skillLevelLabel = 'Skill level';

  export const lessonsPlaceholder = '{lessonsReplace}';
  export const lessonsRemainingButton = '{lessonsReplace} Lessons Remaining';
  export const gradeButton = 'Grade Lesson';
  export const contactStudentButton = 'Contact Student';
}

export namespace BuyMoreLessonsModalComponent {
  export enum FieldKey {
    LessonRate = 'lessonRate',
    LessonDuration = 'lessonDuration',
  }

  export const fieldNames = {
    [FieldKey.LessonRate]: 'lessonRate',
    [FieldKey.LessonDuration]: 'lessonDuration',
  };

  export const fieldLabels = {
    [FieldKey.LessonRate]: 'Lesson rate',
    [FieldKey.LessonDuration]: 'Lesson duration',
  };

  export const placeholders = {
    [FieldKey.LessonRate]: '0.00',
    [FieldKey.LessonDuration]: 'Select lesson duration',
  };

  export const ids = {
    [FieldKey.LessonRate]: 'lessonRate',
    [FieldKey.LessonDuration]: 'lessonDuration',
  };

  export interface BuyMoreLessonsErrors {
    [FieldKey.LessonRate]?: string;
  }

  export const errorMessages: BuyMoreLessonsErrors = {
    [FieldKey.LessonRate]: 'Enter an amount.',
  };

  export const title = 'Send Reminder';
  export const displayNamePlaceholder = '{displayNameReplace}';
  export const instrumentPlaceholder = '{instrumentReplace}';
  /* tslint:disable-next-line:max-line-length */
  export const description = 'Send {displayNameReplace} a reminder to buy more {instrumentReplace} lessons at this rate per lesson';
  export const amountPlaceholder = '{amountReplace}';
  export const youGet = 'You get ${amountReplace}';
  export const cancelButton = 'Cancel';
  export const sendButton = 'Send';
}

export namespace PreLaunchInstructorDashboardComponent {
  export const profileCompletionSectionTitle = 'Profile Completion';
  export const profileSectionTitle = 'View Profile';
  export const descriptionText = 'Your profile is';
  export const percentPlaceholder = '{percentReplace}';
  export const completionText = '{percentReplace}% completed';
  export const editProfileLink = 'Edit your profile';
  export const viewProfileLink = 'View your profile';
  export const viewJobsLink = 'View available jobs';
}

export namespace PreLaunchStudentDashboardComponent {
  export const studentSectionTitle = 'Student Details';
  export enum Labels {
    Instrument = 'Instrument',
    SkillLevel = 'Skill level',
    Name = 'Name',
    Age= 'Age',
    LessonDuration = 'Lesson duration',
    LessonPlace = 'Place for lessons',
  }
}

export namespace StudentCardFormComponent {
  export enum Labels {
    Instrument = 'Instrument',
    SkillLevel = 'Skill level',
    Name = 'Name',
    Age= 'Age',
    LessonDuration = 'Lesson duration',
    LessonPlace = 'Place for lessons',
  }

  export enum Placeholders {
    StudentName = 'Student\'s name',
    StudentAge = 'Student\'s age',
    Instrument = 'Select instrument',
    SkillLevel = 'Select level',
    LessonPlace = 'Select place for lessons',
    LessonDuration = 'Select lessons duration'
  }

  export enum FieldKey {
    StudentName = 'name',
    StudentAge = 'age',
    Instrument = 'instrument',
    SkillLevel = 'skillLevel',
    Distance = 'miles',
    LessonPlace = 'lessonPlace',
    LessonDuration = 'lessonDuration',
  }

  export const FieldNames = {
    [FieldKey.StudentName]: 'name',
    [FieldKey.StudentAge]: 'age',
    [FieldKey.Instrument]: 'instrument',
    [FieldKey.SkillLevel]: 'skillLevel',
    [FieldKey.LessonPlace]: 'lessonPlace',
    [FieldKey.LessonDuration]: 'lessonDuration',
  };

  export enum Ids {
    StudentName = 'name',
    StudentAge = 'age',
    Instrument = 'instrument',
    SkillLevel = 'skillLevel',
    LessonPlace = 'lessonPlace',
    LessonDuration = 'lessonDuration',
  }

  export enum ButtonText {
    AddSubmit = 'Add',
    EditSubmit = 'Save Changes',
    Cancel = 'Cancel'
  }
}
