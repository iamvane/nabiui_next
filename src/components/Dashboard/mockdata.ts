export const studentMockDataEmpty = [];

export const studentMockData = [
  {
    user: {
      id: '0',
      email: 'rox@gmail.com',
      role: 'student',
      firstName: 'Roxana',
      lastName: 'Gallardo',
      birthday: '1996-07-12T04:00:00.000Z'
    },
    students: [
      {
        name: 'Roxana',
        age: 22,
        instrument: 'piano',
        skillLevel: 'beginner',
        placeForLessons: 'home',
        lessonDuration: '45 mins'
      }
    ],
    lessonsRemaining: 4
  },
  {
    user: {
      id: '0',
      email: 'kev@gmail.com',
      role: 'student',
      firstName: 'Kevin',
      lastName: 'Charles',
      birthday: '1992-05-26T04:00:00.000Z'
    },
    students: [
      {
        name: 'Kevin',
        age: 22,
        instrument: 'piano',
        skillLevel: 'beginner',
        placeForLessons: 'home',
        lessonDuration: '45 mins'
      }
    ],
    lessonsRemaining: 0
  },
  {
    user: {
      id: '0',
      email: 'sam@gmail.com',
      role: 'parent',
      firstName: 'Sam',
      lastName: 'Flood',
      birthday: '1960-05-26T04:00:00.000Z'
    },
    students: [
      {
        name: 'Zoe',
        age: 6,
        instrument: 'piano',
        skillLevel: 'beginner',
        placeForLessons: 'home',
        lessonDuration: '45 mins'
      },
      {
        name: 'Esme',
        age: 9,
        instrument: 'piano',
        skillLevel: 'beginner',
        placeForLessons: 'home',
        lessonDuration: '45 mins'
      }
    ],
    lessonsRemaining: 5
  },
];
