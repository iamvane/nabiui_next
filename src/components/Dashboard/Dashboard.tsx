import * as React from 'react';
import { connect } from 'react-redux';
import {
  Action,
  Dispatch
} from 'redux';
import { RouteComponentProps } from 'react-router';
import { Link } from 'react-router-dom';

import {
  CircularProgress,
  Typography } from '@material-ui/core';

import { StoreState } from 'redux/store';
import { UserType } from 'redux/models/UserModel';
import SnackBar from 'components/common/SnackBar';
import { AnnouncementConstants } from 'components/common/constants/Announcement';
import { LoggedInPageTemplate } from 'components/common/Templates/LoggedInPageTemplate';
import { Routes } from 'components/common/constants/Routes';
import { fetchUser } from 'redux/actions/UserActions';
import PreLaunchInstructorDashboard from 'components/Dashboard/InstructorDashboard/PreLaunchInstructorDashboard';
import PreLaunchStudentDashboard from 'components/Dashboard/StudentDashboard/PreLaunchStudentDashboard';
import {
  DashboardComponent,
  PreLaunchInstructorDashboardComponent as constants
} from 'components/Dashboard/constants';
import { Role } from 'components/common/constants/Registration';
import SectionTitle from 'components/common/SectionTitle';
import { page } from 'utils/analytics';
import { InstructorType } from 'redux/models/InstructorModel';

interface State {
  showSnackbar: boolean;
}

interface StateProps {
  user: UserType;
  isRequesting: boolean;
  firstName: string;
}

interface DispatchProps {
  fetchUser: () => void;
}

interface OwnProps {
}

interface Props extends
  OwnProps,
  RouteComponentProps<{}>,
  StateProps,
  DispatchProps {}

export class Dashboard extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      showSnackbar: false
    };
  }

  componentDidMount() {
    this.props.fetchUser();
    if (this.props.location.state && this.props.location.state.redirectedFrom === Routes.BuildRequest) {
      this.setState({
        showSnackbar: true
      });
    }

    const userId = this.props.user ? this.props.user.email : 'anonymous';

    const analiticsProps = {
      userId,
      properties: {
        referrer: document.referrer
      }
    };
    page('Dashboard', analiticsProps);
  }

  public closeSnackbar = () => this.setState({ showSnackbar: false });

  render() {

    return (
      <React.Fragment>
        {this.props.isRequesting ?
        <div className="nabi-text-center">
          <CircularProgress />
        </div> :
        <React.Fragment>
          <LoggedInPageTemplate
            sidebarContent={
              this.props.user.role === Role.instructor ?
              <React.Fragment>
                 <SectionTitle text="Things to do" />
                  <img
                    /* tslint:disable-next-line:max-line-length */
                    src="https://nabimusic.s3.us-east-2.amazonaws.com/things-to-do-nabi-music-lessons-boston-new-york.png"
                    className="dashboard-view-profile-image"
                  />
                  <Typography><Link to={Routes.BuildProfile}>{constants.editProfileLink}</Link></Typography>
                  {/* tslint:disable-next-line:max-line-length */}
                  <Typography><Link to={`profile/${this.props.user.profile && (this.props.user.profile as InstructorType).instructorId}`}>{constants.viewProfileLink}</Link></Typography>
                  <Typography><a target="_blank" href="https://www.facebook.com/pg/nabimusicinfo/jobs">{constants.viewJobsLink}</a></Typography>
              </React.Fragment>
                :
              <React.Fragment>
                <SectionTitle text={`Hello ${this.props.firstName ? this.props.firstName : ''}`} />
                <img
                  className="nabi-responsive-image"
                  alt="logo"
                  src={require('../../../assets/images/linkedin-pic.jpg')}
                />
            </React.Fragment>
            }
            mainContent={
              this.props.user.role === Role.instructor ?
              <PreLaunchInstructorDashboard /> : <PreLaunchStudentDashboard />}
            pageTitle={DashboardComponent.pageTitle}
          />
          <SnackBar
            isOpen={this.state.showSnackbar}
            message={AnnouncementConstants.requestSentMessage}
            handleClose={this.closeSnackbar}
            variant="success"
            hideIcon={true}
          />
        </React.Fragment>}
      </React.Fragment>
    );
  }
}

function mapStateToProps(state: StoreState, _ownProps: OwnProps): StateProps {
  const {
    actions: {
      fetchUser: {
        isRequesting,
      }
    }
  } = state.user;

  return {
    user: state.user.user,
    firstName: state.user.user.firstName,
    isRequesting
  };
}

const mapDispatchToProps = (
  dispatch: Dispatch<Action>
): DispatchProps => ({
  fetchUser: () => dispatch(fetchUser())
});

export default connect(mapStateToProps, mapDispatchToProps)(Dashboard);
