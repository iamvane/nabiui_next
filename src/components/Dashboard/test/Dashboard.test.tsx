import * as React from 'react';
import { shallow } from 'enzyme';

import { Dashboard } from 'components/Dashboard/Dashboard';
import { Role } from 'components/common/constants/Registration';
import { UserType } from 'redux/models/UserModel';
import { accountInfo } from 'components/AccountInfo/test/mockData';

describe('Dashboard', () => {
  const match = {
    isExact: true,
    params: {
      step: 'fooId'
    },
    path: '/build-request/:id',
    url: '/build-request/fooId'
  };

  const location = {
    state: null,
    hash: '',
    pathname: '/build-request/fooId',
    search: ''
  };

  let mock: any = jest.fn();

  const props = {
    match,
    location,
    history: mock,
  };

  const mockUser: UserType = {
    ...accountInfo,
    id: 'fooId',
    firstName: 'foo',
    lastName: 'bar',
    email: 'oreo',
    password: '',
    zipCode: '',
    role: Role.instructor,
    displayName: ''
  };

  const  wrapper = shallow(
    <Dashboard
      firstName=""
      isRequesting={false}
      fetchUser={() => jest.fn}
      user={mockUser}
      {...props}
    />);

  it('Matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
