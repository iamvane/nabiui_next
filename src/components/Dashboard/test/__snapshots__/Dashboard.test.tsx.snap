// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dashboard
    fetchUser={[Function]}
    firstName=""
    history={[MockFunction]}
    isRequesting={false}
    location={
      Object {
        "hash": "",
        "pathname": "/build-request/fooId",
        "search": "",
        "state": null,
      }
    }
    match={
      Object {
        "isExact": true,
        "params": Object {
          "step": "fooId",
        },
        "path": "/build-request/:id",
        "url": "/build-request/fooId",
      }
    }
    user={
      Object {
        "displayName": "",
        "email": "oreo",
        "firstName": "foo",
        "gender": "",
        "id": "fooId",
        "lastName": "bar",
        "lat": "",
        "lng": "",
        "location": "",
        "middleName": "",
        "password": "",
        "phone": Object {
          "isVerified": false,
          "phoneNumber": "",
        },
        "phoneNumber": "",
        "referralToken": "",
        "role": "instructor",
        "zipCode": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <UNDEFINED>
        <Unknown
          mainContent={<Connect(PreLaunchInstructorDashboard) />}
          pageTitle="Dashboard"
          sidebarContent={
            <UNDEFINED>
              <SectionTitle
                text="Things to do"
              />
              <img
                className="dashboard-view-profile-image"
                src="https://nabimusic.s3.us-east-2.amazonaws.com/things-to-do-nabi-music-lessons-boston-new-york.png"
              />
              <WithStyles(Typography)>
                <Link
                  to="/build-profile"
                >
                  Edit your profile
                </Link>
              </WithStyles(Typography)>
              <WithStyles(Typography)>
                <Link
                  to="profile/undefined"
                >
                  View your profile
                </Link>
              </WithStyles(Typography)>
              <WithStyles(Typography)>
                <a
                  href="https://www.facebook.com/pg/nabimusicinfo/jobs"
                  target="_blank"
                >
                  View available jobs
                </a>
              </WithStyles(Typography)>
            </UNDEFINED>
          }
        />
        <SnackBar
          handleClose={[Function]}
          hideIcon={true}
          isOpen={false}
          message="Invite friends and family. For every person you invite you will get a FREE lesson!"
          variant="success"
        />
      </UNDEFINED>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Unknown
            mainContent={<Connect(PreLaunchInstructorDashboard) />}
            pageTitle="Dashboard"
            sidebarContent={
              <UNDEFINED>
                <SectionTitle
                  text="Things to do"
                />
                <img
                  className="dashboard-view-profile-image"
                  src="https://nabimusic.s3.us-east-2.amazonaws.com/things-to-do-nabi-music-lessons-boston-new-york.png"
                />
                <WithStyles(Typography)>
                  <Link
                    to="/build-profile"
                  >
                    Edit your profile
                  </Link>
                </WithStyles(Typography)>
                <WithStyles(Typography)>
                  <Link
                    to="profile/undefined"
                  >
                    View your profile
                  </Link>
                </WithStyles(Typography)>
                <WithStyles(Typography)>
                  <a
                    href="https://www.facebook.com/pg/nabimusicinfo/jobs"
                    target="_blank"
                  >
                    View available jobs
                  </a>
                </WithStyles(Typography)>
              </UNDEFINED>
            }
          />,
          <SnackBar
            handleClose={[Function]}
            hideIcon={true}
            isOpen={false}
            message="Invite friends and family. For every person you invite you will get a FREE lesson!"
            variant="success"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "mainContent": <Connect(PreLaunchInstructorDashboard) />,
            "pageTitle": "Dashboard",
            "sidebarContent": <UNDEFINED>
              <SectionTitle
                text="Things to do"
              />
              <img
                className="dashboard-view-profile-image"
                src="https://nabimusic.s3.us-east-2.amazonaws.com/things-to-do-nabi-music-lessons-boston-new-york.png"
              />
              <WithStyles(Typography)>
                <Link
                  to="/build-profile"
                >
                  Edit your profile
                </Link>
              </WithStyles(Typography)>
              <WithStyles(Typography)>
                <Link
                  to="profile/undefined"
                >
                  View your profile
                </Link>
              </WithStyles(Typography)>
              <WithStyles(Typography)>
                <a
                  href="https://www.facebook.com/pg/nabimusicinfo/jobs"
                  target="_blank"
                >
                  View available jobs
                </a>
              </WithStyles(Typography)>
            </UNDEFINED>,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleClose": [Function],
            "hideIcon": true,
            "isOpen": false,
            "message": "Invite friends and family. For every person you invite you will get a FREE lesson!",
            "variant": "success",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <UNDEFINED>
          <Unknown
            mainContent={<Connect(PreLaunchInstructorDashboard) />}
            pageTitle="Dashboard"
            sidebarContent={
              <UNDEFINED>
                <SectionTitle
                  text="Things to do"
                />
                <img
                  className="dashboard-view-profile-image"
                  src="https://nabimusic.s3.us-east-2.amazonaws.com/things-to-do-nabi-music-lessons-boston-new-york.png"
                />
                <WithStyles(Typography)>
                  <Link
                    to="/build-profile"
                  >
                    Edit your profile
                  </Link>
                </WithStyles(Typography)>
                <WithStyles(Typography)>
                  <Link
                    to="profile/undefined"
                  >
                    View your profile
                  </Link>
                </WithStyles(Typography)>
                <WithStyles(Typography)>
                  <a
                    href="https://www.facebook.com/pg/nabimusicinfo/jobs"
                    target="_blank"
                  >
                    View available jobs
                  </a>
                </WithStyles(Typography)>
              </UNDEFINED>
            }
          />
          <SnackBar
            handleClose={[Function]}
            hideIcon={true}
            isOpen={false}
            message="Invite friends and family. For every person you invite you will get a FREE lesson!"
            variant="success"
          />
        </UNDEFINED>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Unknown
              mainContent={<Connect(PreLaunchInstructorDashboard) />}
              pageTitle="Dashboard"
              sidebarContent={
                <UNDEFINED>
                  <SectionTitle
                    text="Things to do"
                  />
                  <img
                    className="dashboard-view-profile-image"
                    src="https://nabimusic.s3.us-east-2.amazonaws.com/things-to-do-nabi-music-lessons-boston-new-york.png"
                  />
                  <WithStyles(Typography)>
                    <Link
                      to="/build-profile"
                    >
                      Edit your profile
                    </Link>
                  </WithStyles(Typography)>
                  <WithStyles(Typography)>
                    <Link
                      to="profile/undefined"
                    >
                      View your profile
                    </Link>
                  </WithStyles(Typography)>
                  <WithStyles(Typography)>
                    <a
                      href="https://www.facebook.com/pg/nabimusicinfo/jobs"
                      target="_blank"
                    >
                      View available jobs
                    </a>
                  </WithStyles(Typography)>
                </UNDEFINED>
              }
            />,
            <SnackBar
              handleClose={[Function]}
              hideIcon={true}
              isOpen={false}
              message="Invite friends and family. For every person you invite you will get a FREE lesson!"
              variant="success"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "mainContent": <Connect(PreLaunchInstructorDashboard) />,
              "pageTitle": "Dashboard",
              "sidebarContent": <UNDEFINED>
                <SectionTitle
                  text="Things to do"
                />
                <img
                  className="dashboard-view-profile-image"
                  src="https://nabimusic.s3.us-east-2.amazonaws.com/things-to-do-nabi-music-lessons-boston-new-york.png"
                />
                <WithStyles(Typography)>
                  <Link
                    to="/build-profile"
                  >
                    Edit your profile
                  </Link>
                </WithStyles(Typography)>
                <WithStyles(Typography)>
                  <Link
                    to="profile/undefined"
                  >
                    View your profile
                  </Link>
                </WithStyles(Typography)>
                <WithStyles(Typography)>
                  <a
                    href="https://www.facebook.com/pg/nabimusicinfo/jobs"
                    target="_blank"
                  >
                    View available jobs
                  </a>
                </WithStyles(Typography)>
              </UNDEFINED>,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "handleClose": [Function],
              "hideIcon": true,
              "isOpen": false,
              "message": "Invite friends and family. For every person you invite you will get a FREE lesson!",
              "variant": "success",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": Symbol(react.fragment),
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
